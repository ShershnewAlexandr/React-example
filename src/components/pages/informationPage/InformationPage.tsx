import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Grid } from '@material-ui/core';
import { useTranslation } from 'react-i18next';
import { routes } from '../../../utils/constants';
import ContactsFormContainer from '../../../containers/contacts/ContactsFormContainer';
import EducationFormContainer from '../../../containers/education/EducationFormContainer';
import ExperienceFormContainer from '../../../containers/experience/ExperienceFormContainer';
import LanguageFormContainer from '../../../containers/languages/LanguageFormContainer';
import ProgressBarContainer from '../../../containers/progressBar/ProgressBarContainer';
import HelperProgressBarContainer from '../../../containers/progressBar/HelperProgressBarContainer';
import Loader from '../../loaders/Loader';
import './ContactsPageStyles.css';

interface InformationPageProps {
  isLoading: boolean;
}

function InformationPage(props: InformationPageProps) {
  const { t } = useTranslation();
  return props.isLoading ? (
    <Loader />
  ) : (
    <div className="container">
      <Grid container direction="column" justify="flex-start">
        <Grid item xs={3}>
          <h1>
            <Switch>
              <Route
                path={routes.CONTACTS}
                render={props => t('contacts.contacts')}
              />
              <Route
                path={routes.EDUCATION}
                render={props => t('education.education')}
              />
              <Route
                path={routes.EXPERIENCE}
                render={props => t('experience.experience')}
              />
              <Route
                path={routes.LANGUAGES}
                render={props => t('languages.languages')}
              />
            </Switch>
          </h1>
        </Grid>
        <Grid item xs={12}>
          <Grid container direction="row" spacing={5}>
            <Grid item xs={3}>
              <div className="black-font">
                <span>
                  <Switch>
                    <Route
                      path={routes.CONTACTS}
                      render={props => t('contacts.addContactInfo')}
                    />
                    <Route
                      path={routes.EDUCATION}
                      render={props => t('education.addEducationInfo')}
                    />
                    <Route
                      path={routes.EXPERIENCE}
                      render={props => t('experience.addExperienceInfo')}
                    />
                    <Route
                      path={routes.LANGUAGES}
                      render={props => t('languages.addLanguagesInfo')}
                    />
                  </Switch>
                </span>
              </div>
              <hr />
              <div className="grey-font">
                <span>
                  <Switch>
                    <Route
                      path={routes.CONTACTS}
                      render={props => t('contacts.provideBasicInfo')}
                    />
                    <Route
                      path={routes.EDUCATION}
                      render={props => t('education.provideBasicInfo')}
                    />
                    <Route
                      path={routes.EXPERIENCE}
                      render={props => t('experience.provideBasicInfo')}
                    />
                    <Route
                      path={routes.LANGUAGES}
                      render={props => t('languages.provideBasicInfo')}
                    />
                  </Switch>
                </span>
              </div>
            </Grid>
            <Grid item xs={9}>
              <Switch>
                <Route
                  path={routes.CONTACTS}
                  render={props => <ContactsFormContainer />}
                />
                <Route
                  path={routes.EDUCATION}
                  render={props => <EducationFormContainer />}
                />
                <Route
                  path={routes.EXPERIENCE}
                  render={props => <ExperienceFormContainer />}
                />
                <Route
                  path={routes.LANGUAGES}
                  render={props => <LanguageFormContainer />}
                />
              </Switch>
              <HelperProgressBarContainer>
                <ProgressBarContainer />
              </HelperProgressBarContainer>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </div>
  );
}

export default InformationPage;
