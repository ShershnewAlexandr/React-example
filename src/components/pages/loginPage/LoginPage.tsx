import React from 'react';
import LoginFormContainer from '../../../containers/login/LoginFormContainer';
import './LoginPageStyle.css';

function LoginPage() {
  return (
    <div className="login-container">
      <LoginFormContainer />
    </div>
  );
}

export default LoginPage;
