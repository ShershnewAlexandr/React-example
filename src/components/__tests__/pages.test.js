import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../../setupTests';
import LoginPage from '../pages/loginPage/LoginPage';
import InformationPage from '../pages/informationPage/InformationPage';

jest.mock('../../store/index');
jest.mock('i18next', () => ({
  use: function() {
    return this;
  },
  init: jest.fn(),
}));
jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: str => str,
  }),
}));

describe('pages', () => {
  const loginPage = shallow(<LoginPage />);
  const informationPageLoading = shallow(<InformationPage isLoading={true} />);
  const informationPage = shallow(<InformationPage isLoading={false} />);

  it('LoginPage snapshot', () => {
    expect(toJson(loginPage)).toMatchSnapshot();
  });

  it('InformationPageLoading snapshot', () => {
    expect(toJson(informationPageLoading)).toMatchSnapshot();
  });

  it('InformationPage snapshot', () => {
    expect(toJson(informationPage)).toMatchSnapshot();
  });
});
