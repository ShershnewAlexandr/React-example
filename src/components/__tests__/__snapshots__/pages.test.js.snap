// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages InformationPage snapshot 1`] = `
<div
  className="container"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
    direction="column"
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={3}
    >
      <h1>
        <Switch>
          <Route
            path="/information/contacts"
            render={[Function]}
          />
          <Route
            path="/information/education"
            render={[Function]}
          />
          <Route
            path="/information/experience"
            render={[Function]}
          />
          <Route
            path="/information/languages"
            render={[Function]}
          />
        </Switch>
      </h1>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        direction="row"
        spacing={5}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={3}
        >
          <div
            className="black-font"
          >
            <span>
              <Switch>
                <Route
                  path="/information/contacts"
                  render={[Function]}
                />
                <Route
                  path="/information/education"
                  render={[Function]}
                />
                <Route
                  path="/information/experience"
                  render={[Function]}
                />
                <Route
                  path="/information/languages"
                  render={[Function]}
                />
              </Switch>
            </span>
          </div>
          <hr />
          <div
            className="grey-font"
          >
            <span>
              <Switch>
                <Route
                  path="/information/contacts"
                  render={[Function]}
                />
                <Route
                  path="/information/education"
                  render={[Function]}
                />
                <Route
                  path="/information/experience"
                  render={[Function]}
                />
                <Route
                  path="/information/languages"
                  render={[Function]}
                />
              </Switch>
            </span>
          </div>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={9}
        >
          <Switch>
            <Route
              path="/information/contacts"
              render={[Function]}
            />
            <Route
              path="/information/education"
              render={[Function]}
            />
            <Route
              path="/information/experience"
              render={[Function]}
            />
            <Route
              path="/information/languages"
              render={[Function]}
            />
          </Switch>
          <Connect(HelperProgressBarContainer)>
            <withRouter(Connect(ProgressBarContainer)) />
          </Connect(HelperProgressBarContainer)>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`pages InformationPageLoading snapshot 1`] = `<Loader />`;

exports[`pages LoginPage snapshot 1`] = `
<div
  className="login-container"
>
  <Connect(ReduxForm) />
</div>
`;
